### PR TITLE
Substitute header, body and footer view names in template with variables

### PR DESCRIPTION
**Pull request by misquith**
This change allows users to extend the header, body and footer
views provided by ember-table.

For e.g. the user can extend BodyTableContainer with some custom features. ember-table's bodyTableContainerView can then be overridden to use the enhanced body.

### CODE REVIEW

**Comment by misquith** [comment]
@Addepar/pants @Addepar/fire

File: addon/components/ember-table.js [hunk 1]
```diff
@@ -75,6 +75,15 @@ StyleBindingsMixin, ResizeHandlerMixin, {
   // through ctrl/cmd-click or shift-click).
   selectionMode: 'single',
 
+  //Name of header view used in the table. This property can be overridden to specify a custom view. The custom view must extend HeaderTableContainer
```
Comment (embooglement) [inline addon/components/ember-table.js]: Spaces after the start of a comment.

**Comment by embooglement** [comment]
This should have some unit tests probably. Basically just checking that setting the property actually changes the output.

**Comment by misquith** [comment]
@Addepar/pants @Addepar/fire

File: tests/unit/ember-table-test.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    'view:multi-item-collection',
+    'view:header-cell',
+    'view:lazy-table-block',
+    'view:footer-table-container',
+    'view:table-block',
+    'view:scroll-panel'
+  ]
+});
+
+test('Overriding the bodyTableContainerView property results in the specified being rendered', function(assert) {
```
Comment (embooglement) [inline tests/unit/ember-table-test.js]: Missing word. Should be `results in the specified view being rendered`.

Comment (embooglement) [inline tests/unit/ember-table-test.js]: Same with the rest of these.

File: dist/ember-table.js [hunk 3]
```diff
@@ -196,6 +196,15 @@ var define, requireModule, require, requirejs;
       // through ctrl/cmd-click or shift-click).
       selectionMode: 'single',
 
+      // Name of header view used in the table. This property can be overridden to specify a custom view. The custom view must extend HeaderTableContainer
+      headerTableContainerView: 'header-table-container',
+
+      // Name of the body view used in the table.  This property can be overridden to specify a custom view. The custom view must extend BodyTableContainer
+      bodyTableContainerView: 'body-table-container',
+
+      // Name of the footer view used in the table.  This property can be overridden to specify a custom view. The custom view must extend BodyTableContainer
```
Comment (embooglement) [inline dist/ember-table.js]: Should this be "must extend FooterTableContainer"

File: addon/components/ember-table.js [hunk 4]
```diff
@@ -75,6 +75,15 @@ StyleBindingsMixin, ResizeHandlerMixin, {
   // through ctrl/cmd-click or shift-click).
   selectionMode: 'single',
 
+  // Name of header view used in the table. This property can be overridden to specify a custom view. The custom view must extend HeaderTableContainer
+  headerTableContainerView: 'header-table-container',
+
+  // Name of the body view used in the table.  This property can be overridden to specify a custom view. The custom view must extend BodyTableContainer
+  bodyTableContainerView: 'body-table-container',
+
+  // Name of the footer view used in the table.  This property can be overridden to specify a custom view. The custom view must extend BodyTableContainer
```
Comment (embooglement) [inline addon/components/ember-table.js]: Should this be "must extend FooterTableContainer"

**Comment by embooglement** [comment]
LGTM after comments

**Pull request merged**